### PR TITLE
Included more parameters in realtime server http requests to the controller.

### PR DESCRIPTION
**Pull request by mcvsubbu**
In preparation for auto-tuning of consuming segment allocation in multi-tenant
realtime cluser:

The realtime segment completion requests posted to the controller now include
the following parameters as appropriate:
        memory used
        number of rows consumed
        build time
        wait time to acquire the build semaphore.

Broadened the scope of SegmentFileAndOffset class (renamed it to SegmentBuildDescriptor) to
include more than just the file name and offset. It now serves as a return value of
buildSegmentInternal() method, and has the build time and wait time as well.

### CODE REVIEW

File: pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java [hunk 1]
```diff
@@ -561,17 +578,17 @@ private File makeSegmentDirPath() {
     return new File(_resourceDataDir, _segmentZKMetadata.getSegmentName());
   }
 
-  protected String buildSegmentForCommit(long buildTimeLeaseMs) {
+  // Modifies _segmentBuildDescriptor if we have do not have a valid built segment file.
```
Comment (npawar) [inline pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java]: confusing comment "have do not have"

Comment (mcvsubbu) [reply pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java]: fixed. It should be "do not have"

File: pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java [hunk 2]
```diff
@@ -561,17 +578,17 @@ private File makeSegmentDirPath() {
...
+  protected void buildSegmentForCommit(long buildTimeLeaseMs) {
     try {
-      if (_segmentFileAndOffset != null) {
-        if (_segmentFileAndOffset.getOffset() == _currentOffset) {
+      if (_segmentBuildDescriptor != null) {
```
[the remark this answers is not in the document]
Comment (mcvsubbu) [reply pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java]: OK, combined.

File: pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java [hunk 3]
```diff
@@ -611,16 +637,14 @@ protected String buildSegmentInternal(boolean forCommit) {
...
       try {
         converter.build(_segmentVersion, _serverMetrics);
       } catch (Exception e) {
         segmentLogger.error("Could not build segment", e);
         FileUtils.deleteQuietly(tempSegmentFolder);
         return null;
       }
-      final long buildEndTime = now();
-      segmentLogger.info("Successfully built segment in {} ms", (buildEndTime - buildStartTime));
+      segmentLogger.info("Successfully built segment in {} ms", (now() - lockAquireTimeMillis));
```
[the remark this answers is not in the document]
Comment (mcvsubbu) [reply pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java]: Done

**Pull request merged**